packages/aws-cdk/lib/api/hotswap/lambda-functions.ts

```diff
@@ -281,3 +295,55 @@ class LambdaFunctionHotswapOperation implements HotswapOperation {
     return Promise.all(operations);
   }
 }
+
+/**
+ * Compress a string as a file, returning a promise for the zip buffer
+ * https://github.com/archiverjs/node-archiver/issues/342
+ */
+export function zipString(fileName: string, rawString: string): Promise<Buffer> {
+  return new Promise((resolve, reject) => {
+    const buffs: Buffer[] = [];
+
+    const converter = new Writable();
+
+    converter._write = (chunk: Buffer, _: string, cb: () => void) => {
+      buffs.push(chunk);
+      process.nextTick(cb);
+    };
+
+    converter.on('finish', () => {
+      resolve(Buffer.concat(buffs));
+    });
+
+    const archive = archiver('zip');
+
+    archive.on('error', (err) => {
+      reject(err);
+    });
+
+    archive.pipe(converter);
+
+    archive.append(rawString, {
+      name: fileName,
+      date: new Date('1980-01-01T00:00:00.000Z'), // Add date to make resulting zip file deterministic
+    });
+
+    void archive.finalize();
+  });
+}
+
+/**
+ * Get file extension from Lambda runtime string.
+ * We use this extension to create a deployment package from Lambda inline code.
+ */
+function getExtensionFromRuntime(runtime: string): string {
```

Comment:
1. This function should be first in the file (it's called first).
2. Let's rename it to `determineCodeFileExtFromRuntime()`.